src/interface.rs

```diff
@@ -0,0 +1,131 @@
+use super::{Error, Miniconf};
+use core::fmt::Write;
+use heapless::{consts, String};
+
+/// Represents an incoming MQTT packet.
+pub struct Message<'a> {
```

Comment:
Let's add first-class support for minimq for the `Message` and `Response` properties - specifically, ```
impl From<(&[u8], &[minimq::Property])> for Message { ... }
impl Response {
    to_minimq(&self) -> (&[u8], &[minimq::Property));
}
```